src/autocomplete/databases/clickhouse/grammar/ClickHouseParser.g4

```diff
@@ -337,16 +338,31 @@ explainStatement
     | EXPLAIN ESTIMATE notInsertStatement   # ExplainEstimateStatement
     ;
 
+// REVOKE statement
+
+revokeStatement
+    : REVOKE clusterClause? (GRANT OPTION FOR)? privilegeList ON accessSubjectIdentifier FROM (userExpressionList | ALL | ALL EXCEPT userExpressionList)
+    | REVOKE clusterClause? (ADMIN OPTION FOR)? roleExpressionList FROM (userOrRoleExpressionList | ALL | ALL EXCEPT userOrRoleExpressionList)
+    ;
+
+userExpressionList
+    : userIdentifier (COMMA userIdentifier)*
+    ;
+
+roleExpressionList
+    : roleIdentifier (COMMA roleIdentifier)*
+    ;
+
 // GRANT statement
 
 grantStatement
-    : GRANT clusterClause? privilegeList ON grantSubjectIdentifier TO userOrRoleExpressionList (WITH GRANT OPTION)? (WITH REPLACE OPTION)?
-    | GRANT clusterClause? roleIdentifier TO userOrRoleExpressionList (WITH ADMIN OPTION)? (WITH REPLACE OPTION)?
-    | GRANT CURRENT GRANTS ((LPAREN privilegeList ON grantSubjectIdentifier RPAREN) | ON grantSubjectIdentifier) TO userOrRoleExpressionList (WITH GRANT OPTION)? (WITH REPLACE OPTION)?
+    : GRANT clusterClause? (privilegeList ON accessSubjectIdentifier) (COMMA privilegeList ON accessSubjectIdentifier)* TO userOrRoleExpressionList (WITH GRANT OPTION)? (WITH REPLACE OPTION)?
```

Comment:
nit: I think we can move out `WITH REPLACE OPTION`, it is copy pasted multiple times